joint_state_topic_hardware_interface/test/example_position.cpp

```diff
@@ -0,0 +1,211 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+
+#include <control_msgs/action/follow_joint_trajectory.hpp>
+
+std::shared_ptr<rclcpp::Node> node;
+bool common_goal_accepted = false;
+rclcpp_action::ResultCode common_resultcode = rclcpp_action::ResultCode::UNKNOWN;
+int common_action_result_code = control_msgs::action::FollowJointTrajectory_Result::SUCCESSFUL;
+
+std::vector<double> desired_goals = { 0, -0.5, 0.5, 0.0 };
+unsigned int ct_goals_reached = 0;
+
+void common_goal_response(rclcpp_action::ClientGoalHandle<control_msgs::action::FollowJointTrajectory>::SharedPtr future)
+{
+  RCLCPP_DEBUG(node->get_logger(), "common_goal_response time: %f", rclcpp::Clock().now().seconds());
+  auto goal_handle = future.get();
+  if (!goal_handle)
+  {
+    common_goal_accepted = false;
+    printf("Goal rejected\n");
+  }
+  else
+  {
+    common_goal_accepted = true;
+    printf("Goal accepted\n");
+  }
+}
+
+void common_result_response(
+    const rclcpp_action::ClientGoalHandle<control_msgs::action::FollowJointTrajectory>::WrappedResult& result)
+{
+  printf("common_result_response time: %f\n", rclcpp::Clock(RCL_ROS_TIME).now().seconds());
+  common_resultcode = result.code;
+  common_action_result_code = result.result->error_code;
+  switch (result.code)
+  {
+    case rclcpp_action::ResultCode::SUCCEEDED:
+      printf("Action goal succeeded\n");
+      break;
+    case rclcpp_action::ResultCode::ABORTED:
+      printf("Action goal was aborted\n");
+      return;
+    case rclcpp_action::ResultCode::CANCELED:
+      printf("Action goal was canceled\n");
+      return;
+    default:
+      printf("Action goal: Unknown result code\n");
+      return;
+  }
+}
+
+void common_feedback(rclcpp_action::ClientGoalHandle<control_msgs::action::FollowJointTrajectory>::SharedPtr,
+                     const std::shared_ptr<const control_msgs::action::FollowJointTrajectory::Feedback> feedback)
+{
+  std::cout << "feedback->desired: positions: " << feedback->desired.positions.at(0);
+  std::cout << ", velocities: " << feedback->desired.velocities.at(0) << std::endl;
+
+  std::cout << "feedback->actual: positions: " << feedback->actual.positions.at(0);
+  std::cout << std::endl;
+
+  if (ct_goals_reached < desired_goals.size())
+  {
+    if (fabs(feedback->actual.positions[0] - desired_goals.at(ct_goals_reached)) < 0.1)
```

Comment:
This seems like a large tolerance for mock_hardware. Does it need to be this big?